### ISSUE TITLE
Regression in 3.8.1: ApartmentAttribute no longer works when applied to an assembly

### ISSUE THREAD

**Issue by thecontrarycat** [nunit/nunit#2426]
Create a test project containing the following test:

```
[assembly:Apartment(ApartmentState.STA)]
namespace TestProject
{
    [TestFixture]
    public class TestFixture
    {
        [Test]
        public void Test()
        {
            if (Thread.CurrentThread.GetApartmentState() != ApartmentState.STA)
            {
                throw new ThreadStateException("The current threads apartment state is not STA");
            }
        }
    }
}
```

Compile it against NUnit 3.7.1 - Test Passes
Compile it against NUnit 3.8.1 - Test Fails

Compile it against NUnit 3.8.1 but with the `[Apartment]` attribute on the TestFixture - Test Passes
Compile it against NUnit 3.8.1 but with the `[Apartment]` attribute on the Test - Test Passes

Note that this bug is present in 3.8.0 as well as 3.8.1

**Comment by CharliePoole** [comment]
I'm suspicious that this could be a side effect of the fix to #2361

**Comment by CharliePoole** [comment]
This is different from but probably related to #2388

**Comment by CharliePoole** [comment]
I'm going away on a family trip for a couple of weeks so I'm taking my name off this to free it up for somebody else to work on.

**Comment by rinubaby** [comment]
We've already upgraded to 3.8.1 (was on 2.6.2 earlier), since we have a large number of tests, we are being blocked by this issue, any timeline as to when the next release (fix) will be available? Also, seeing occasionally that test runs are getting hung. Is this only a 3.8.1 problem, wondering if going back to 3.7.1 would fix the hanging issue as well. Thanks.

**Comment by ChrisMaddock** [comment]
Both these issues were regressions in 3.8.x - so 3.7.1 should work for you until the next release.

I don't believe anyone's currently working on this issue - so as to a fix it really depends when someone picks it up and submits a pull request.

**Comment by rinubaby** [comment]
I have a fix for this issue, can I do a pull request so that you can look at it? I am getting access denied.

**Comment by CharliePoole** [comment]
Anyone can do a pull request. Did you fork to your own account first?

**Comment by rinubaby** [comment]
Thanks Charlie, have opened a pull request.

**Comment by rprouse** [comment]
I tested with 3.9.0-dev-04639 and this has not been fixed with our recent parallel fixes.

This is possibly a duplicate of #2388
